src/client/pages/CountryHome/Repository/Panel/Actions/hooks/useOnSaveFile.ts

```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react'
+
+import { CountryIso } from 'meta/area'
+
+import { useAppDispatch } from 'client/store'
+import { RepositoryActions } from 'client/store/ui/repository'
+import { useCountryRouteParams } from 'client/hooks/useRouteParams'
+
+type Returned = () => Promise<void>
+
+export const useOnSaveFile = (): Returned => {
+  const dispatch = useAppDispatch()
+  const { assessmentName, cycleName, countryIso } = useCountryRouteParams<CountryIso>()
+
+  return useCallback<Returned>(async () => {
+    const saveParams = { assessmentName, cycleName, countryIso }
+    dispatch(RepositoryActions.createRepositoryItem(saveParams))
```

Comment:
The repository item we want to create is fetched from the redux store in the action using `getState`